### ISSUE TITLE
Bug with multiplexed async connections in combination with pipeline (redis::pipe)?

### ISSUE THREAD

**Issue by brocaar** [redis-rs/redis-rs#1078]
I'm storing metrics data in Redis. Depending the time-range, this will retrieve one or multiple keys for which using `redis::pipe()` to `HGETALL` these keys in one go. E.g. these keys could be `metrics:20240310`, `metrics:20240311`, `metrics:20240312`. This was all working very well until I migrated the async connections to async multiplexed connections after upgrading to v0.25 and seeing the deprecation notices.

Suddenly I got errors like:

> Response was of incompatible type - TypeError: "Response type not hashmap compatible" (response was string-data('"field1"'))

When using  `get_async_connection()`, all was working fine again. This is very simple to reproduce, see the below code example. 

### Code example to reproduce the issue

```rust
use std::collections::HashMap;

/* In Redis:

redis> HSET myhash field1 "Hello"
(integer) 1
redis> HSET myhash field2 "World"
(integer) 1
redis> HGETALL myhash
1) "field1"
2) "Hello"
3) "field2"
4) "World"

redis>
*/

#[tokio::main]
async fn main() {
    let client = redis::Client::open("redis://127.0.0.1/").unwrap();

    // This works
    let mut con = client.get_async_connection().await.unwrap();

    let mut pipe = redis::pipe();
    pipe.cmd("HGETALL").arg("myhash");

    let res: Vec<HashMap<String, String>> = pipe.query_async(&mut con).await.unwrap();

    println!("{:?}", res);

    // This fails
    // The code is exactly the same, but we are using a multiplexed async connection
    let mut con = client.get_multiplexed_async_connection().await.unwrap();

    let mut pipe = redis::pipe();
    pipe.cmd("HGETALL").arg("myhash");

    let res: Vec<HashMap<String, String>> = pipe.query_async(&mut con).await.unwrap();

    println!("{:?}", res);
}
```

This outputs:

```
warning: use of deprecated method `redis::Client::get_async_connection`: aio::Connection is deprecated. Use client::get_multiplexed_async_connection instead.
 --> src/main.rs:6:26
  |
6 |     let mut con = client.get_async_connection().await.unwrap();
  |                          ^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(deprecated)]` on by default

warning: `redis-test` (bin "redis-test") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 1.66s
     Running `target/debug/redis-test`
[{"field1": "Hello", "field2": "World"}]
thread 'main' panicked at src/main.rs:20:78:
called `Result::unwrap()` on an `Err` value: Response was of incompatible type - TypeError: "Response type not hashmap compatible" (response was string-data('"field1"'))
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Interestingly, this would work (multiple `HGETALL` commands in one pipeline):

```rust
    let mut con = client.get_multiplexed_async_connection().await.unwrap();

    let mut pipe = redis::pipe();
    pipe.cmd("HGETALL").arg("myhash");
    pipe.cmd("HGETALL").arg("myhash");

    let res: Vec<HashMap<String, String>> = pipe.query_async(&mut con).await.unwrap();

    println!("{:?}", res);
```

**Comment by jaymell** [comment]
Thanks for reporting. It appears this broke in https://github.com/redis-rs/redis-rs/commit/70dfb95713fbcfe944a9262189c2adf23c12977c

**Comment by nihohit** [comment]
@jaymell that's what I thought, too. on it.

**Comment by nihohit** [comment]
Fixed in 0.25.2, thanks for reporting this!

**Comment by brocaar** [comment]
Thanks @jaymell and @nihohit for your quick feedback and fix. I can confirm that v0.25.2 solves this issue.
